File: JavaPOO/src/ec/edu/utpl/poo/Student.java
package ec.edu.utpl.poo;

public class Student {
    // Atributo de clase
    public static String university;
    // Atributo de instancia
    private String name;
    private int age;
    private int semester;
    private String career;

    // Constructor

    public Student(String name, int age, int semester) {
        this.name = name;
        validarDato(age);
        this.age = age;
        validarDato(semester);
        this.semester = semester;
        setUniversity("UTPL");
    }

    public Student(String name, int age, int semester, String career) {
        this.name = name;
        validarDato(age);
        this.age = age;
        validarDato(semester);
        this.semester = semester;
        this.career = career;
        setUniversity("UTPL");
    }

    // Getters y setters atributos de instancia

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getAge() {
        return age;
    }

    public void setAge(int age) {
        this.age = age;
    }

    public int getSemester() {
        return semester;
    }

    public void setSemester(int semester) {
        this.semester = semester;
    }

    public String getCareer() {
        return career;
    }

    public void setCareer(String carreer) {
        this.career = carreer;
    }

    // Getter y setter atributos de clase

    public static String getUniversity() {
        return university;
    }

    public static void setUniversity(String university) {
        Student.university = university;
    }

    // Validación de datos
    private void validarDato(int valor) {
        if (valor <= 0) {
            throw new IllegalArgumentException("Denominador no puede ser 0 o negativo");
        }
    }
}

File: PythonPOO/Student.py
class Student:
    # Atributo de clase
    university = "UTPL"

    # Constructor
    def __init__(self, name: str = 'N/A', age: int = 0, semester: int = 0, career: str = 'N/A'):
        # Atributos de instancia
        self.__name = name
        self.__career = career
        try:
            # Validación de datos
            if age <= 0 or semester <= 0:
                raise Exception("\nDato incorrecto, introduzca número mayor a 0")
            self.__age = age
            self.__semester = semester
        except ValueError:
            print("")


    # Getters y setters de atributos de instancia

    @property
    def name(self):
        return self.__name

    @name.setter
    def name(self, name):
        self.__name = name

    @property
    def age(self):
        return self.__age

    @age.setter
    def age(self, age):
        self.__age = age

    @property
    def semester(self):
        return self.__semester

    @semester.setter
    def semester(self, semester):
        self.__semester = semester

    @property
    def career(self):
        return self.__career

    @career.setter
    def career(self, career):
        self.__career = career

    # Getters y setters de atributos de clase

    @classmethod
    def getUniversity(cls):
        return cls.university

    @classmethod
    def setUniversity(cls, university):
        cls.university = university
